lib/Endpoint.js

```diff
@@ -71,6 +71,67 @@ class Endpoint {
     return this._started
   }
 
+  async pause (cold) {
+    if (!this._started || this._paused) {
+      if (this._resuming) {
+        console.warn('Tried to pause endpoint whilst busy resuming')
+      }
+
+      return this._paused
+    }
+
+    this._paused = new Promise((resolve, reject) => {
```

Comment:
could this._pause just = Promise.all(ops)

Reply:
Aye it could! Remnants of old attempts. 👍 

Might be able to change `resume()` to that too.

Reply:
Ah. There was a reason it's like this. See 96368f7.